components/resource_metering/src/localstorage.rs

```diff
@@ -49,35 +25,5 @@ pub struct LocalStorage {
 #[derive(Clone)]
 pub struct LocalStorageRef {
     pub id: usize,
-    pub storage: LocalStorage,
-}
-
-/// Register a channel to notify thread creation events.
-pub fn register_storage_chan_tx(tx: Sender<LocalStorageRef>) {
-    STORAGE_CHANS.lock().unwrap().push(tx)
-}
-
-#[cfg(test)]
-mod tests {
-    use super::*;
-    use crossbeam::channel::unbounded;
-
-    #[test]
-    fn test_storage_chan() {
-        let (tx, rx) = unbounded();
-        register_storage_chan_tx(tx);
-        STORAGE.with(|_| {}); // Just to trigger registration.
-        std::thread::spawn(move || {
-            STORAGE.with(|_| {});
-        })
-        .join()
-        .unwrap();
-        let mut count = 0;
-        while let Ok(r) = rx.try_recv() {
-            assert_ne!(r.id, 0);
-            count += 1;
-        }
-        // This value may be greater than 2 if other test threads access `STORAGE` in parallel.
-        assert!(count >= 2);
-    }
+    pub storage: LocalStorage, // TODO: change to shared_ptr
```

Comment:
Does this TODO mean to use `SharedTagPtr` directly instead? We will add more fields to `LocalStorage` in the scan-row PR, so here we still need to refer to the entire LocalStorage instead of a single tag.

Reply:
`SharedTagPtr` contains an Arc. Can we use the same `Arc` to wrap other fields together? Of course, will rename `SharedTagPtr`.

Reply:
Yes, we can put them in LocalStorage, of course they can also be placed under `SharedTagPtr` (need rename). See [#11085 L76-L79](https://github.com/tikv/tikv/pull/11085/files#diff-8a4ce4dc3080cfb7d9bf6ae0f53d8e877a4fedc8321298ed79b695914373ed17R76-R79)

Reply:
Then we can simply `cleanup` recorder.